File: problemas/iniciante/1079/1079.lua
n = tonumber(io.read(), 10)
for _ = 1, n do
  entrada = {}
  for numero in io.read():gmatch("%S+") do
    entrada[#entrada + 1] = numero
  end

  x = tonumber(entrada[1])
  y = tonumber(entrada[2])
  z = tonumber(entrada[3])
  media = (2.0 * x + 3.0 * y + 5.0 * z) / 10.0

  print(string.format("%.1f", media))
end

File: problemas/iniciante/1079/1079.hs
import Text.Printf

calcula :: Int -> IO ()
calcula 0 = return ()
calcula n = do
  entrada <- getLine
  let [a, b, c] = map read (words entrada) :: [Double]
  let media = (2 * a + 3 * b + 5 * c) / 10
  printf "%.1f\n" media
  calcula (n - 1)

main :: IO ()
main = do
  n <- readLn :: IO Int
  calcula n